src/core/public/chrome/ui/global_search/page_item.tsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiBreadcrumb, EuiHighlight, EuiSimplifiedBreadcrumbs } from '@elastic/eui';
+import {
+  ChromeNavLink,
+  ChromeRegistrationNavLink,
+  NavGroupItemInMap,
+} from 'opensearch-dashboards/public';
+import React from 'react';
+
+interface Props {
+  link: ChromeRegistrationNavLink & ChromeNavLink & { navGroup: NavGroupItemInMap };
+  search: string;
+  callback?: () => void;
+  renderBreadcrumbs?: (breadcrumbs: EuiBreadcrumb[]) => EuiBreadcrumb[];
+}
+
+export const GlobalSearchPageItem = ({ link, search, callback, renderBreadcrumbs }: Props) => {
+  let breadcrumbs: EuiBreadcrumb[] = [];
+
+  const appId = link.id.toLowerCase();
+  const isLanding = appId.endsWith('landing');
+  const text = (
+    <EuiHighlight search={search} highlightAll={true}>
+      {isLanding ? `${link.navGroup.title} ${link.title}` : link.title}
+    </EuiHighlight>
+  );
+
+  const isOverviewPage = appId.endsWith('overview');
+  if (isOverviewPage && link.category) {
+    breadcrumbs.push({ text: link.category.label });
+  }
+
+  if (link.parentNavLinkId) {
+    const parentNavLinkTitle = link.navGroup.navLinks.find(
+      (navLink) => navLink.id === link.parentNavLinkId
+    )?.title;
+    if (parentNavLinkTitle) {
+      breadcrumbs.push({
+        text: (
+          <EuiHighlight search={search} highlightAll={true}>
+            {parentNavLinkTitle}
+          </EuiHighlight>
+        ),
+      });
+    }
+  }
+
+  const onNavItemClick = () => {
+    callback?.();
+  };
+
+  if (renderBreadcrumbs) {
+    breadcrumbs = renderBreadcrumbs(breadcrumbs);
+  }
```

Comment:
Prefer to move this line to line 41 and handle with all breadcrumb stuff in a `if else`. Not readable enough since `breadcrumb` is re-assigned in separate lines.